backend/backend/core/tools.py

```diff
@@ -7,7 +8,7 @@
 from langchain_core.tools import Tool, BaseTool
 from langchain_experimental.utilities import PythonREPL
 from langchain_community.tools import TavilySearchResults
-from github import Github
+from github import Github, GithubException
```

Comment:
The import of 'github' has been moved to the top. This is a good practice for readability and maintainability.